### DISCUSSION TITLE
qt teamtalk. Host manager

### DISCUSSION THREAD

**Discussion by peter1384** [BearWare/TeamTalk5#2268] [Ideas]
### Description

@CoBC hi
i want to ask would it be possible for you to also move the connect button which is in the main screen of the host manager to the context menu of the servers? because context menu means all things depend on that thing should be there
and connect button needs a specification of knowing which server to be connect to.
so the connect should be moved to the context menus, and only add a new server and impoart a .tt file button should be in the main screen because these button's are not dependent of any server or anything else.
What do you say?

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by CoBC** [comment]
Not sure on real utility to add this action in context menu. I see 2 options:
* Just keep as it is currently
* Remove "connect" button and just let enter and double click have action to connect
Currently, when you tab to "connect" button and you try to activate it, no action happened, don't know behaviour with mouse.

@bear101 Any opinion?

**Comment by peter1384** [comment]
@CoBC oh. I really understand you.
But, don't worry. Please freely remove it from the main screen and add it to the context menu because of the below reasons.
1: doing this will remove one unnesisery button from the main screen, so it will be even more simplyfied, and make this screen more clean and standard ui, which was the main point of this new host manager.
2: adding it to the context menu is a standard ui that everyone does that (that's actually why something called context menu has been made), context menu means when you open it, you want to see all of the related actions for that thing, with out making the main screen too much full.
and now that you've recoded the host manager and this new host manager looks really clean and standard, please also remove this connect button from the main screen and add the connect action to the context menu and wright a proper void for it like other options, so users can focus a server and press application and press the connect, or they can just press enter on it to connect, depending on how they want to do it.
And it will make this new host manager as standard as possible

**Comment by peter1384** [comment]
@CoBC also, the option 2 that you said is incorrect.
• Remove "connect" button and just let enter and double click have action to connect
this is incorrect, because with a standard ui, every doable action should have a shortcut and an option, wether that be a button in the main screen which is incorrect in this case of connect, or a context menu option which is correct for this case which should be happen by a dubble klick or pressing enter on an item

**Comment by bear101** [comment]
@CoBC when you press "Connect" button initially then you connect to the latest server. I think this behaviour is fine, since it's fast and what user most likely wants to do

**Comment by bear101** [comment]
@peter1384 please write in Discussions either Q/A or Ideas. If it's an issue then we'll create the issue.

**Comment by peter1384** [comment]
@bear101 the connect button doesn't even work to connect to a laytist server, but ok. We'll not remove it, but i'll open a pull request to add the connect option to the context menus because some people and I need that, And it won't be bad for anyone.
